.github/docker_images/scripts/create-compiler-env.sh

```diff
@@ -0,0 +1,191 @@
+        echo "$version_output" | head -n1 | grep -oE '[0-9]+\.[0-9]+(\.[0-9]+)?' | head -n1 | cut -d. -f1
+    elif version_output=$("$compiler" -dumpversion 2>/dev/null); then
+        echo "$version_output" | cut -d. -f1
+    else
+        return 1
+    fi
+}
+
+# Function to discover versioned compilers from filesystem
+discover_versioned_compilers() {
+    local compiler_base="$1"  # e.g., "gcc" or "clang"
+    local cxx_base="$2"       # e.g., "g++" or "clang++"
+    local versions=()
+    
+    # Search for versioned compilers in standard paths
+    for search_path in "${SEARCH_PATHS[@]}"; do
+        if [[ -d "$search_path" ]]; then
+            # Find all versioned compilers (e.g., gcc-11, clang-14)
+            while IFS= read -r -d '' compiler_path; do
+                local basename_comp=$(basename "$compiler_path")
+                # Extract version from filename (e.g., gcc-11 -> 11, gcc-4.1 -> 4.1)
+                if [[ "$basename_comp" =~ ^${compiler_base}-([0-9]+(\.[0-9]+)?)$ ]]; then
+                    local version="${BASH_REMATCH[1]}"
+                    local cxx_path="${search_path}/${cxx_base}-${version}"
+                    
+                    # Check if corresponding C++ compiler exists
+                    if [[ -x "$cxx_path" ]]; then
+                        versions+=("$version")
+                    fi
+                fi
+            done < <(find "$search_path" -maxdepth 1 -name "${compiler_base}-[0-9]*" \( -type f -o -type l \) \
+                -executable -print0 2>/dev/null)
+        fi
+    done
+    
+    # Remove duplicates and sort (only if versions array is not empty)
+    if [[ ${#versions[@]} -gt 0 ]]; then
+        printf '%s\n' "${versions[@]}" | sort -uV
+    fi
+}
+
+# Function to create a compiler setup script
+create_compiler_script() {
+    local script_name="$1"
+    local cc_compiler="$2"
+    local cxx_compiler="$3"
+    local display_name="$4"
+    
+    local script_path="${COMPILER_ENV_DIR}/${script_name}"
+    
+    echo "Creating compiler setup script: ${script_path}"
+    
+    cat > "${script_path}" << EOF
+# Compiler environment setup for ${display_name}
+export CC=${cc_compiler}
+export CXX=${cxx_compiler}
+echo "Compiler environment set to ${display_name}"
+echo "CC=\$CC"
+echo "CXX=\$CXX"
+EOF
+    
+    echo "Created ${script_path}"
+}
+
+# Function to detect and create GCC compiler scripts
+detect_gcc_compilers() {
+    echo "Detecting GCC compilers..."
+    
+    # Discover versioned GCC compilers
+    local versions
+    mapfile -t versions < <(discover_versioned_compilers "gcc" "g++")
+    
+    if [[ ${#versions[@]} -gt 0 ]]; then
+        for version in "${versions[@]}"; do
+            if [[ -n "$version" ]]; then
+                create_compiler_script "setup-gcc-${version}.sh" "gcc-${version}" "g++-${version}" "GCC ${version}"
+            fi
+        done
+    fi
+    
+    # Handle base GCC compiler
```

Comment:
Would technically create scripts for the same compiler if `gcc` is symbolically linked to any of the versioned ones.

Probably not a blocker.